lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4011,7 +4013,7 @@ namespace {
       Optional<ImportedName> correctSwiftName;
       ImportedName importedName;
 
-      if (!decl->isAnonymousStructOrUnion()) {
+      if (!decl->isAnonymousStructOrUnion() && !decl->getDeclName().isEmpty()) {
```

Comment:
`FieldDecl::isAnonymousStructOrUnion()` actually bails out early with a `false` return if the name isn't empty, so you could replace the call to it with the empty-name check (here and elsewhere) instead of joining them with `&&`. (If you do that, though, you should probably add a comment explaining which situations result in an empty name—otherwise the intent will be lost.)

Reply:
Offline, you said you preferred to keep this implementation because the intent was clearer. I'm fine with that.